### PR TITLE
Mtl 2643 emv 3 ds api for auth only

### PR DESCRIPTION
**Pull request by shawn9001**

### CODE REVIEW

File: source/docs/guides/EMV_3D_secure.md [hunk 1]
```diff
@@ -406,14 +406,9 @@ Payments response:
...
-## Use your own process
-
-Some large merchants complete the Visa Secure, MasterCard SecureCode, or AMEX SafeKey certification to handle 
-authentication on their own side. These merchants can use their existing Visa Secure, SecureCode, or SafeKey authentication 
-process, and send the results of the bank authentication to us with their standard transaction request. To do 
-this, the merchant must integrate using a server-to-server type connection.
+## Use your own 3D Secure MPI
 
-Note: This option must be enabled by us. Contact Customer Care if you want to use this method.
+If you have your own 3D Secure MPI and have no need to use our integrated MPI solution you can pass through method.  This allows you to use your existing Visa Secure, SecureCode, or SafeKey authentication process, and send the results of the 3DSecure authentication to us with their standard transaction request.
```
Comment (Bambora-Dave-Robinson) [inline source/docs/guides/EMV_3D_secure.md]: Grammar on this line; maybe missing the word 'use' near the end of the first sentence?

File: source/docs/guides/EMV_3D_secure.md [hunk 2]
```diff
@@ -564,6 +559,338 @@ trnApproved=1
 &3DSecureStatus=Success
 ```
 
+## EMV 3D Secure API
+
+The EMV 3D Secure API provides a set of end points that you can utilize to perform 3DSecure authentication without immediately processing a payment.  This allows the card holder and credit card number to be authenticated and potentially stored or later referenced in a payment transaction.  Our Payments API supports accepting the 3DS Session Data a token to the card data utilized in a previously authenticated 3D Secure transaction.
```
Comment (Bambora-Dave-Robinson) [inline source/docs/guides/EMV_3D_secure.md]: Need a comma after 'to be authenticated'

Comment (Bambora-Dave-Robinson) [reply source/docs/guides/EMV_3D_secure.md]: supposed to be 'as a token to the card data'?

File: source/docs/guides/EMV_3D_secure.md [hunk 3]
```diff
@@ -564,6 +559,338 @@ trnApproved=1
...
+|message_category|Enum|Set to a value of ‘PaymentAuthentication’ when the 3DS check is being performed before authorizing a charge to the card.  Use ‘NonPaymentAuthentication’ when there will be no immediate authorization such as storing a card to a profile for later use.|
+|amount|Numeric|The amount to be charged in the payment associated to this authentication.|
+|card.number|String|The credit card number to authenticate against|
+|card.expiry.month|Number|The two digit month of the card expiration date.|
+|card.expiry.year|Number|The four digit year of the card expiration date.|
+|token|String|Single-use token id associated to the card to authenticate|
+|payment_profile.customer_code|String|The Secure Payment Profile Customer Code to process the authentication against|
+|payment_profile.card_id|Number|The Card Id to process the authentication against.  This is an optional field, where if not provided the default card will be referenced.|
+|reference|String|Reference field to associate with the transaction. The API does not eforce uniqueness but it is recommended that a unique transaction reference be passed in this field to support.|
```
Comment (Bambora-Dave-Robinson) [inline source/docs/guides/EMV_3D_secure.md]: 'enforce'

Comment (Bambora-Dave-Robinson) [reply source/docs/guides/EMV_3D_secure.md]: 'to support' what?

File: source/docs/guides/EMV_3D_secure.md [hunk 4]
```diff
@@ -564,6 +559,338 @@ trnApproved=1
...
+        "protocol_version": "2.2"
+    },
+    "error": null,
+    "created_datetime_utc": "2021-12-14T02:11:50.45"
+}
+```
+
+### Processing a Payment Using a 3DS Sesstion Data Token
+
+A credit card that has been successfully authenticated through the EMV 3DS API can be later referenced to complete a payment without needing to resubmit the credit card number and expiry. The value of the 'threeDS_session_data' parameter returned in the authentication response can be used a card number token when calling the Payment API to charge the card.
```
Comment (Bambora-Dave-Robinson) [inline source/docs/guides/EMV_3D_secure.md]: Comma needed after 'complete a payment'.
Supposed to be 'used as a card number'?

File: source/docs/guides/EMV_3D_secure.md [hunk 5]
```diff
@@ -564,6 +559,338 @@ trnApproved=1
...
+
+In the payment request we must set the 'payment_method' to a value of '3d_secure' then pass a '3d_secure' object containing the 'threeDS_session_data' as the card token and indicate if 'complete' should be true or false.
```
Comment (Bambora-Dave-Robinson) [inline source/docs/guides/EMV_3D_secure.md]: Comma needed after the first '3d_secure'.
Probably one after 'the card token' as well, but not as critical

File: source/docs/guides/EMV_3D_secure.md [hunk 6]
```diff
@@ -596,7 +923,9 @@ trnApproved=1
     - Place the value of '3d_session_data' in the URI rather than the value of 'md'
     - Pass a value of 'cres' in the body rather than parameter 'pa_res'.
 
-## Status details 
+## 3D Secure Status
+
+The 3D Secure Status returned in the authentication response indicates if the card holder was successfully authenticated and if there was a libility shift.
```
Comment (Bambora-Dave-Robinson) [inline source/docs/guides/EMV_3D_secure.md]: 'liability'
Comma after 'authenticated'

File: source/docs/guides/EMV_3D_secure.md (hunk 1 above)
Comment (shawn9001) [reply source/docs/guides/EMV_3D_secure.md]: updated

File: source/docs/guides/EMV_3D_secure.md (hunk 2 above)
Comment (shawn9001) [reply source/docs/guides/EMV_3D_secure.md]: updated

File: source/docs/guides/EMV_3D_secure.md (hunk 3 above)
Comment (shawn9001) [reply source/docs/guides/EMV_3D_secure.md]: updated

File: source/docs/guides/EMV_3D_secure.md (hunk 4 above)
Comment (shawn9001) [reply source/docs/guides/EMV_3D_secure.md]: updated

File: source/docs/guides/EMV_3D_secure.md (hunk 5 above)
Comment (shawn9001) [reply source/docs/guides/EMV_3D_secure.md]: updated

File: source/docs/guides/EMV_3D_secure.md (hunk 6 above)
Comment (shawn9001) [reply source/docs/guides/EMV_3D_secure.md]: updated

**Comment by shawn9001** [comment]
Made recommended updates or changed wording

File: source/docs/guides/EMV_3D_secure.md [hunk 7]
```diff
@@ -561,7 +561,7 @@ trnApproved=1
 
 ## EMV 3D Secure API
 
-The EMV 3D Secure API provides a set of end points that you can utilize to perform 3DSecure authentication without immediately processing a payment.  This allows the card holder and credit card number to be authenticated and potentially stored or later referenced in a payment transaction.  Our Payments API supports accepting the 3DS Session Data a token to the card data utilized in a previously authenticated 3D Secure transaction.
+The EMV 3D Secure API provides a set of end points that you can utilize to perform 3DSecure authentication without immediately processing a payment.  This can be useful in instances where may want to authenticate a card holder before storing their credentials to a customer profile or for authenticating a transaction that may be processed at a later time.  Our Payments API supports accepting the 3DS Session Data token of a previously authenticated 3D Secure transaction as a reference to the card data and 3D Secure authentication results.
```
Comment (Bambora-Dave-Robinson) [inline source/docs/guides/EMV_3D_secure.md]: where [you] may want
comma before 'or' after 'profile'
comma before 'and 3DS Secure' after 'card data'

File: source/docs/guides/EMV_3D_secure.md [hunk 8]
```diff
@@ -869,9 +869,9 @@ curl --location --request GET 'https://api.na.bambora.com/v1/EMV3DS/MDBiYmI5NTYt
 
 ### Processing a Payment Using a 3DS Sesstion Data Token
 
-A credit card that has been successfully authenticated through the EMV 3DS API can be later referenced to complete a payment without needing to resubmit the credit card number and expiry. The value of the 'threeDS_session_data' parameter returned in the authentication response can be used a card number token when calling the Payment API to charge the card.
+A credit card that has been successfully authenticated through the EMV 3DS API can be later referenced to complete a payment, without needing to resubmit the credit card number and expiry. The value of the 'threeDS_session_data' parameter returned in the authentication response can be used as a token to the card data and 3D Secure results when sumbitting a request to the Payment API.
```
Comment (Bambora-Dave-Robinson) [inline source/docs/guides/EMV_3D_secure.md]: comma before 'when submitting'
'submitting' spelled wrong

File: source/docs/guides/EMV_3D_secure.md [hunk 9]
```diff
@@ -406,14 +406,9 @@ Payments response:
...
-## Use your own process
-
-Some large merchants complete the Visa Secure, MasterCard SecureCode, or AMEX SafeKey certification to handle 
-authentication on their own side. These merchants can use their existing Visa Secure, SecureCode, or SafeKey authentication 
-process, and send the results of the bank authentication to us with their standard transaction request. To do 
-this, the merchant must integrate using a server-to-server type connection.
+## Payment API - Passthrough Method
 
-Note: This option must be enabled by us. Contact Customer Care if you want to use this method.
+If you have your own 3D Secure MPI and have no need to use our integrated MPI solution you can use the Passthrough Method. This allows you to use your existing Visa Secure, SecureCode, or SafeKey authentication process, and send the results of the 3DSecure authentication to us with your payment request.
```
Comment (Bambora-Dave-Robinson) [inline source/docs/guides/EMV_3D_secure.md]: comma after 'MPI solution'

File: source/docs/guides/EMV_3D_secure.md (hunk 7 above)
Comment (shawn9001) [reply source/docs/guides/EMV_3D_secure.md]: Updates

File: source/docs/guides/EMV_3D_secure.md (hunk 8 above)
Comment (shawn9001) [reply source/docs/guides/EMV_3D_secure.md]: Updated

File: source/docs/guides/EMV_3D_secure.md (hunk 9 above)
Comment (shawn9001) [reply source/docs/guides/EMV_3D_secure.md]: Updated

File: data/na/swagger/1-0-5.yaml [hunk 10]
```diff
@@ -921,6 +923,25 @@ definitions:
         type: string
         maxLength: 20
         description: 'The Visa Checkout payment request ID.'
+  3d_secure:
+    description: "Payment with a 3D Secure Session Data token.  Reference a previously proessed 3D Secure transaction
```
Comment (Bambora-Dave-Robinson) [inline data/na/swagger/1-0-5.yaml]: 'processed' spelled wrong

File: data/na/swagger/1-0-5.yaml [hunk 11]
```diff
@@ -921,6 +923,25 @@ definitions:
         type: string
         maxLength: 20
         description: 'The Visa Checkout payment request ID.'
+  3d_secure:
+    description: "Payment with a 3D Secure Session Data token.  Reference a previously proessed 3D Secure transaction 
+      processed through the EMV 3DS API.  This will apply the associated card number, card expiry, and 3DSecure results
```
Comment (Bambora-Dave-Robinson) [inline data/na/swagger/1-0-5.yaml]: need a space in '3D Secure' for consistency

File: source/docs/guides/EMV_3D_secure.md [hunk 12]
```diff
@@ -406,14 +406,9 @@ Payments response:
...
-## Use your own process
-
-Some large merchants complete the Visa Secure, MasterCard SecureCode, or AMEX SafeKey certification to handle 
-authentication on their own side. These merchants can use their existing Visa Secure, SecureCode, or SafeKey authentication 
-process, and send the results of the bank authentication to us with their standard transaction request. To do 
-this, the merchant must integrate using a server-to-server type connection.
+## Payment API - Passthrough Method
 
-Note: This option must be enabled by us. Contact Customer Care if you want to use this method.
+If you have your own 3D Secure MPI and have no need to use our integrated MPI solution, you can use the Passthrough Method. This allows you to use your existing Visa Secure, SecureCode, or SafeKey authentication process, and send the results of the 3DSecure authentication to us with your payment request.
```
Comment (Bambora-Dave-Robinson) [inline source/docs/guides/EMV_3D_secure.md]: '3D Secure' needs a space

File: data/na/swagger/1-0-5.yaml (hunk 10 above)
Comment (shawn9001) [reply data/na/swagger/1-0-5.yaml]: updated

File: data/na/swagger/1-0-5.yaml (hunk 11 above)
Comment (shawn9001) [reply data/na/swagger/1-0-5.yaml]: updated

File: source/docs/guides/EMV_3D_secure.md (hunk 12 above)
Comment (shawn9001) [reply source/docs/guides/EMV_3D_secure.md]: Updated

File: source/docs/guides/EMV_3D_secure.md [hunk 13]
```diff
@@ -561,17 +561,17 @@ trnApproved=1
...
+The EMV 3D Secure API provides a set of end points that you may want to utilize to perform 3D Secure authentication without immediately processing a payment.  This can be useful in instances where may want to authenticate a card holder before storing their credentials to a customer profile, or for authenticating a transaction that may be processed at a later time.  Our Payments API supports accepting the 3DS Session Data token of a previously authenticated 3D Secure transaction as a reference to the card data, and 3D Secure authentication results.
 
 A passcode must be passed in the authorization header when making a request to the EMV 3DS API.  This API shares the same passcode as the Payment API.
 
 Access to this API must be enabled by a member of our support team.  For assistance, you can send a message to Client Services or call 1-888-472-0811.
 
 ### /POST /V1/EMV3DS/AuthRequst
 
-This endpoint is your first touchpoint for validating card via the EMV 3DSecure process; all new requests must start here.
+This endpoint is your first touchpoint for validating card via the EMV 3D Secure process; all new requests must start here.
```
Comment (Bambora-Dave-Robinson) [inline source/docs/guides/EMV_3D_secure.md]: validating [a] card

File: source/docs/guides/EMV_3D_secure.md [hunk 14]
```diff
@@ -561,17 +561,17 @@ trnApproved=1
 
 ## EMV 3D Secure API
 
-The EMV 3D Secure API provides a set of end points that you may want to utilize to perform 3DSecure authentication without immediately processing a payment.  This can be useful in instances where may want to authenticate a card holder before storing their credentials to a customer profile, or for authenticating a transaction that may be processed at a later time.  Our Payments API supports accepting the 3DS Session Data token of a previously authenticated 3D Secure transaction as a reference to the card data, and 3D Secure authentication results.
+The EMV 3D Secure API provides a set of end points that you may want to utilize to perform 3D Secure authentication without immediately processing a payment.  This can be useful in instances where may want to authenticate a card holder before storing their credentials to a customer profile, or for authenticating a transaction that may be processed at a later time.  Our Payments API supports accepting the 3DS Session Data token of a previously authenticated 3D Secure transaction as a reference to the card data, and 3D Secure authentication results.
```
Comment (Bambora-Dave-Robinson) [inline source/docs/guides/EMV_3D_secure.md]: where [you] may want

File: source/docs/guides/EMV_3D_secure.md [hunk 15]
```diff
@@ -934,7 +934,7 @@ The 3D Secure Status returned in the authentication response indicates if the ca
...
 | Unavailable | The 3DS service is unavailable due to technical issues. | If you continue with the transaction there will be no lability shift and there will be risk of chargeback. The transaction and 3DS authentication can be retried at a later time. | No | Transaction processes | Transaction declined message 311 | 7 | 7 |
-| Error | Authentication failed due to an internal error. | If you continue with the transaction there will be no lability shift and there will be risk of chargeback. An unexpected internal error occurred processing the 3DSecure authentication. If the problem persists contact Customer Care. | No | Transaction processes | Transaction declined message 311 | 7 | 7 |
+| Error | Authentication failed due to an internal error. | If you continue with the transaction there will be no lability shift and there will be risk of chargeback. An unexpected internal error occurred processing the 3D Secure authentication. If the problem persists contact Customer Care. | No | Transaction processes | Transaction declined message 311 | 7 | 7 |
```
Comment (Bambora-Dave-Robinson) [inline source/docs/guides/EMV_3D_secure.md]: spelling on liability

**Review by Bambora-Dave-Robinson — APPROVED** [review]

**Pull request merged**